engine/deployment.go

```diff
@@ -17,19 +17,28 @@ func checkRunningDeploymentsConformity(ctx context.Context, l zerolog.Logger, de
 	hasBeenPatched := false
 	for _, d := range deployments {
 		repl := int(*d.Spec.Replicas)
+		// if the current number of replicas is 0, we want to evaluate whether to scale back up to the original replicas
 		if repl == 0 {
+			// if no originalReplicas annotation is found, we assume the desired replicas is 0 to handle a deployment scaled to 0
+			desiredRepl := 0
+			var err error
 			// get the desired number of replicas
-			repl, err := strconv.Atoi(d.Annotations[prefix+originalReplicas])
-			if err != nil {
-				return hasBeenPatched, err
+			if d.Annotations != nil {
+				if val, ok := d.Annotations[prefix+originalReplicas]; ok {
```

Comment:
To avoid one nested if, you can group the two `if` checks:

```go
if val, ok := d.Annotations[prefix+originalReplicas]; ok && d.Annotations != nil {
	desiredRepl, err = strconv.Atoi(val)
	if err != nil {
		return hasBeenPatched, err
	}
}
```